addOns/commonlib/src/test/java/org/zaproxy/addon/commonlib/ResourceidentificationUtilsUnitTest.java

```diff
@@ -0,0 +1,164 @@
+        // Given
+        HttpMessage msg = new HttpMessage();
+        msg.getRequestHeader().setURI(new URI("http://example.com/" + fileName, false));
+        // When / Then
+        assertThat(ResourceIdentificationUtils.isFont(msg), is(equalTo(false)));
+    }
+
+    @ParameterizedTest
+    @ValueSource(strings = {"something.js", "something.jsx"})
+    void shouldReturnTrueWhenRequestUrlSeemsToBeAJavaScriptFile(String fileName)
+            throws HttpMalformedHeaderException, URIException, NullPointerException {
+        // Given
+        msg.getRequestHeader().setURI(new URI("http://example.com/" + fileName, false));
+        // When / Then
+        assertThat(ResourceIdentificationUtils.isJavaScript(msg), is(equalTo(true)));
+    }
+
+    @Test
+    void shouldReturnTrueWhenResponseSeemsToBeAJavaScriptFile()
+            throws HttpMalformedHeaderException, URIException, NullPointerException {
+        // Given
+        msg.setResponseBody("<script>alert('hello');</script>");
+        msg.getRequestHeader().setURI(new URI("http://example.com/any.js", true));
+        msg.getResponseHeader().setHeader(HttpResponseHeader.CONTENT_TYPE, "text/javascript");
+        // When / Then
+        assertThat(ResourceIdentificationUtils.isJavaScript(msg), is(equalTo(true)));
+    }
+
+    @Test
+    void shouldReturnTrueWhenRequestUrlSeemsToBeACssFile()
+            throws HttpMalformedHeaderException, URIException, NullPointerException {
+        // Given
+        msg.getRequestHeader().setURI(new URI("http://example.com/something.css", false));
+        // When / Then
+        assertThat(ResourceIdentificationUtils.isCss(msg), is(equalTo(true)));
+    }
+
+    @Test
+    void shouldReturnTrueWhenResponseSeemsToBeACssFile()
+            throws HttpMalformedHeaderException, URIException, NullPointerException {
+        // Given
+        msg.setResponseBody(
+                "body {background-color: powderblue;}\n"
+                        + "h1 {color: blue;}\n"
+                        + "p {color: red;"
+                        + "margin-left:85.36370249136206%"
+                        + "}");
+        msg.getRequestHeader().setURI(new URI("http://example.com/any.css", true));
+        msg.getResponseHeader().setHeader(HttpResponseHeader.CONTENT_TYPE, "text/css");
+        // When / Then
+        assertThat(ResourceIdentificationUtils.isCss(msg), is(equalTo(true)));
+    }
+
+    @ParameterizedTest
+    @ValueSource(strings = {"hello_world.pdf", "hello_world.png", "hello_world.xlsx"})
+    void shouldReportControlCharactersInBinaryResponse(String fileName)
+            throws HttpMalformedHeaderException, IOException {
+        // Given / When
+        msg = createHttpMessageFromFilename(fileName);
+        // Then
+        assertThat(
+                ResourceIdentificationUtils.responseContainsControlChars(msg), is(equalTo(true)));
+    }
+
+    @ParameterizedTest
+    @ValueSource(strings = {"hello_world.txt", "hello_world.html"})
```

Comment:
Can just hardcode a string without control chars.

Reply:
Again I was trying to be realistic, though with plain text I guess it’s less of an issue.

Reply:
I'll tackle this shortly.